eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClient.RestClient.cs

```diff
@@ -38,314 +38,275 @@ internal PipelineMessage CreateSayHiRequest(string headParameter, string queryPa
         {
-            PipelineMessage message = Pipeline.CreateMessage();
+            HttpMessage message = Pipeline.CreateMessage();
             message.ResponseClassifier = PipelineMessageClassifier200;
-            PipelineRequest request = message.Request;
-            request.Method = "POST";
+            Request request = message.Request;
+            request.Method = new RequestMethod("POST");
             ClientUriBuilder uri = new ClientUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/internalProtocol", false);
-            request.Uri = uri.ToUri();
-            request.Headers.Set("Content-Type", "application/json");
-            request.Headers.Set("Accept", "application/json");
+            request.Uri = uri;
+            request.Headers.Add("Content-Type", "application/json");
+            request.Headers.Add("Accept", "application/json");
             request.Content = content;
-            message.Apply(options);
             return message;
         }
 
-        internal PipelineMessage CreateStillConvenientRequest(RequestOptions options)
+        internal HttpMessage CreateStillConvenientRequest(RequestContext options)
         {
-            PipelineMessage message = Pipeline.CreateMessage();
+            HttpMessage message = Pipeline.CreateMessage();
             message.ResponseClassifier = PipelineMessageClassifier204;
-            PipelineRequest request = message.Request;
-            request.Method = "GET";
+            Request request = message.Request;
+            request.Method = new RequestMethod("GET");
             ClientUriBuilder uri = new ClientUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/stillConvenient", false);
-            request.Uri = uri.ToUri();
-            message.Apply(options);
+            request.Uri = uri;
             return message;
         }
 
-        internal PipelineMessage CreateHeadAsBooleanRequest(string id, RequestOptions options)
+        internal HttpMessage CreateHeadAsBooleanRequest(string id, RequestContext options)
         {
-            PipelineMessage message = Pipeline.CreateMessage();
+            HttpMessage message = Pipeline.CreateMessage();
             message.ResponseClassifier = PipelineMessageClassifier2xxAnd4xx;
-            PipelineRequest request = message.Request;
-            request.Method = "HEAD";
+            Request request = message.Request;
+            request.Method = new RequestMethod("HEAD");
             ClientUriBuilder uri = new ClientUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/headAsBoolean/", false);
             uri.AppendPath(id, true);
-            request.Uri = uri.ToUri();
-            message.Apply(options);
+            request.Uri = uri;
             return message;
         }
 
-        private class Classifier2xxAnd4xx : PipelineMessageClassifier
+        private class Classifier2xxAnd4xx : ResponseClassifier
```

Comment:
Just curious, shouldn't it be Classifier200And204 and only accept 200 and 204? I assume all the code should be the same as what we previous generated in autorest.csharp, is that still makes sense?

Reply:
actually, this should be omitted here

Reply:
This needs further tweak in MGC, I will leave it to next iteration as the change in MGC is pretty big already.